.github/workflows/upgrade_test_workflow_call.yml

```diff
@@ -0,0 +1,358 @@
+        run: npm install -g npm@7
+
+      # Checkout PrestaShop source
+      - uses: actions/checkout@v3
+        name: Checkout PrestaShop repository with source branch/tag ${{ inputs.source_ref }}
+        with:
+          repository: PrestaShop/PrestaShop
+          path: ${{ env.PS_DIR }}
+          ref: ${{ inputs.source_ref }}
+
+      # Certificate
+      - name: Generate a certificate
+        if: startsWith(inputs.source_ref, '8.1') || (inputs.source_ref == 'develop')
+        run: |
+          ## Install MkCert
+          sudo apt install libnss3-tools
+          curl -JLO "https://dl.filippo.io/mkcert/latest?for=linux/amd64"
+          chmod +x mkcert-v*-linux-amd64
+          sudo cp mkcert-v*-linux-amd64 /usr/local/bin/mkcert
+          ## Generate certificate
+          mkcert -key-file ./${{ env.PS_DIR }}/.docker/ssl.key -cert-file ./${{ env.PS_DIR }}/.docker/ssl.crt localhost
+          ## Link certificate to Chrome Trust Store
+          mkdir -p $HOME/.pki/nssdb 
+          certutil -d $HOME/.pki/nssdb -N 
+          certutil -d sql:$HOME/.pki/nssdb -n localhost -A -t "TCu,Cu,Tu" -i ./${{ env.PS_DIR }}/.docker/ssl.crt
+          ## Add self-signed certificate to Chrome Trust Store
+          mkcert -install
+
+      # Workaround until https://github.com/PrestaShop/PrestaShop/issues/29813 is fixed
+      - name: PrestaShop Configuration (Copy of Config API)
+        if: (env.BASE_BRANCH == '8.1.x') || (env.BASE_BRANCH == 'develop')
+        run: cp ./${{ env.PS_DIR }}/app/config/security_test.yml ./${{ env.PS_DIR }}/app/config/security_prod.yml
+
+      # Create shop with Docker
+      - name: Build and run shop with docker
+        working-directory: ${{ env.PS_DIR }}
+        timeout-minutes: 15
+        env:
+          DISABLE_MAKE: 0
+          URL_FO: ${{ (startsWith(inputs.source_ref, '1.7') || startsWith(inputs.source_ref, '8.0')) && 'http://localhost:8001/' || 'https://localhost:8002/' }}
+          VERSION: ${{ startsWith(inputs.source_ref, '1.7') && inputs.php_version || env.VERSION }}
+        run: |
+          echo "    container_name: prestashop_autoupgrade" >> docker-compose.yml
+          USER_ID=$(id -u) GROUP_ID=$(id -g) docker-compose -f docker-compose.yml up -d --build
+          bash -c 'while [[ "$(curl -L -s -o /dev/null -w %{http_code} ${{ env.URL_FO }}en/)" != "200" ]]; do sleep 5; done'
+          docker ps
+
+      # Install upgrade module and run CLI upgrade
+      - uses: actions/checkout@v3
+        name: Checkout Autoupgrade module
+        with:
+          repository: PrestaShop/autoupgrade
+          path: ${{ env.PS_DIR }}/modules/autoupgrade
+          ref: ${{ inputs.upgrade_module_ref }}
+      - name: Composer install in module
+        run: composer install
+        working-directory: ${{ env.PS_DIR }}/modules/autoupgrade
+      - name: Install module via CLI commands
+        run: docker exec -u www-data prestashop_autoupgrade php bin/console prestashop:module install autoupgrade
```

Comment:
before version 1.7.4.0 the console was under /app, this will have to be taken into account if we add older versions